ctapipe/tests/test_atmosphere.py

```diff
@@ -133,3 +133,116 @@ def test_against_reference():
         0,
         atol=1e-5,
     )
+
+
+def test_line_of_sight_integral(table_profile):
+    """
+    Test observer altitude and zenith angle dependence of LoS integral function
+    """
+
+    assert table_profile.line_of_sight_integral(
+        10 * u.km, observer_altitude=2000 * u.m
+    ) < table_profile.line_of_sight_integral(10 * u.km, observer_altitude=1000 * u.m)
+
+    assert table_profile.line_of_sight_integral(
+        10 * u.km, zenith_angle=30 * u.deg
+    ) > table_profile.line_of_sight_integral(10 * u.km, zenith_angle=0 * u.deg)
+
+
+def test_height_overburden_circle(table_profile):
+    """
+    Test that successive application of height to overburden
+    and overburden to height functions return original values
+    """
+
+    # Five-layer atmosphere
+    fit_reference = np.array(
+        [
+            [0.00 * 100000, -140.508, 1178.05, 994186, 0],
+            [9.75 * 100000, -18.4377, 1265.08, 708915, 0],
+            [19.00 * 100000, 0.217565, 1349.22, 636143, 0],
+            [46.00 * 100000, -0.000201796, 703.745, 721128, 0],
+            [106.00 * 100000, 0.000763128, 1, 1.57247e10, 0],
+        ]
+    )
+
+    profile_5 = atmo.FiveLayerAtmosphereDensityProfile.from_array(fit_reference)
+
+    layer_5_heights = u.Quantity([5, 15, 30, 70, 110] * u.km)
+
+    for height in layer_5_heights:
+        circle_height_5_layer = profile_5.height_from_overburden(
+            profile_5.integral(height)
+        )
+
+        assert np.allclose(circle_height_5_layer, height, rtol=0.005)
+
+    # Exponential atmosphere
+    density_model = atmo.ExponentialAtmosphereDensityProfile(
+        scale_height=10 * u.km, scale_density=0.00125 * u.g / u.cm**3
```

Comment:
I think you can import the `DENSITY_UNIT` constant and use it here.

Reply:
All the unit conversions are done